src/vs/editor/contrib/codeAction/browser/codeActionMenu.ts

```diff
@@ -284,84 +256,54 @@ class CodeActionList extends Disposable {
 	}
 
 	public focusPrevious() {
-		if (typeof this.focusedEnabledItem === 'undefined') {
-			this.focusedEnabledItem = this.viewItems[0].index;
-		} else if (this.viewItems.length < 1) {
-			return;
-		}
-
-		const startIndex = this.focusedEnabledItem;
-		let item: ICodeActionMenuItem;
-
-		do {
-			this.focusedEnabledItem = this.focusedEnabledItem - 1;
-			if (this.focusedEnabledItem < 0) {
-				this.focusedEnabledItem = this.viewItems.length - 1;
-			}
-			item = this.viewItems[this.focusedEnabledItem];
-			this.list.setFocus([item.index]);
-			this.currSelectedItem = item.index;
-		} while (this.focusedEnabledItem !== startIndex && item.action.action.disabled);
+		this.list.focusPrevious(1, true, undefined, element => element.kind === CodeActionListItemKind.CodeAction && !element.action.action.disabled);
 	}
 
 	public focusNext() {
-		if (typeof this.focusedEnabledItem === 'undefined') {
-			this.focusedEnabledItem = this.viewItems.length - 1;
-		} else if (this.viewItems.length < 1) {
+		this.list.focusNext(1, true, undefined, element => element.kind === CodeActionListItemKind.CodeAction && !element.action.action.disabled);
+	}
+
+	public acceptSelected() {
+		const focused = this.list.getFocus();
+		if (focused.length === 0) {
 			return;
 		}
 
-		const startIndex = this.focusedEnabledItem;
-		let item: ICodeActionMenuItem;
+		const focusIndex = focused[0];
+		const element = this.list.element(focusIndex);
+		if (element.kind !== CodeActionListItemKind.CodeAction || element.action.action.disabled) {
+			return;
+		}
 
-		do {
-			this.focusedEnabledItem = (this.focusedEnabledItem + 1) % this.viewItems.length;
-			item = this.viewItems[this.focusedEnabledItem];
-			this.list.setFocus([item.index]);
-			this.currSelectedItem = item.index;
-		} while (this.focusedEnabledItem !== startIndex && item.action.action.disabled);
+		this.list.setSelection([focusIndex]);
 	}
 
-	public onEnterSet() {
-		if (typeof this.currSelectedItem === 'number') {
-			this.list.setSelection([this.currSelectedItem]);
+	private onListSelection(e: IListEvent<ICodeActionMenuItem>): void {
+		if (!e.elements.length) {
+			return;
 		}
-	}
 
-	private onListSelection(e: IListEvent<ICodeActionMenuItem>): void {
-		for (const element of e.elements) {
-			if (element.kind === CodeActionListItemKind.CodeAction && !element.action.action.disabled) {
-				this.onDidSelect(element.action);
-			}
+		const element = e.elements[0];
```

Comment:
this went from an array to just the `0` item of an array. Is that expected?

Reply:
Yes we have disabled multiple selections for this list, so for now `e.elements` should only ever contain 1 item